io/src/lib.rs

```diff
@@ -293,4 +328,18 @@ mod tests {
         assert_eq!(fill.len(), 0);
         assert_eq!(fill, &[]);
     }
+
+    #[test]
+    #[cfg(any(feature = "alloc", not(feature = "std")))]
+    fn read_to_limit_greater_than_total_length() {
+        let s = "16-byte-string!!".to_string();
+        let want = s.as_bytes();
+        let mut reader = Cursor::new(&s);
+        let mut buf = vec![];
+
+        // 32 is greater than the reader length.
+        let read = reader.read_to_limit(&mut buf, 32).expect("failed to read to limit");
+        assert_eq!(read, s.len());
+        assert_eq!(&buf, want)
+    }
```

Comment:
I think testing the other case, limit being smaller than the input, would be more valuable since it tests that our DoS protection works.

Reply:
Done.